propagation-stackdriver/src/main/java/zipkin2/propagation/stackdriver/XCloudTraceContextExtractor.java

```diff
@@ -52,24 +53,20 @@ public TraceContextOrSamplingFlags extract(C carrier) {
 
       // Try to parse the trace IDs into the context
       TraceContext.Builder context = TraceContext.newBuilder();
+      long[] traceId = convertHexTraceIdToLong(tokens[0]);
+      if (traceId != null) {
+        boolean traceTrue = true;
+
+        if (tokens.length >= 2) {
```

Comment:
please put a comment here as to what and why so folks who don't look at git log can easily tell.